client/termsetting/handlers/numeric.discrete.ts

```diff
@@ -74,16 +74,17 @@ async function showBinsMenu(self, div: any) {
 	div.append('div').style('padding', '10px').style('text-align', 'center').html('Getting distribution data ...<br/>')
 	try {
 		if (!self.vocabApi) throw `Missing .vocabApi{} [numeric.discrete showBinsMenu()]`
-
+		const tw = await fillTermWrapper({ term: self.term, q: self.q }, self.vocabApi) //the q for the discrete mode may need to be initialized
 		const d = await self.vocabApi.getViolinPlotData(
 			{
-				term: { term: self.term, q: self.q },
+				term: tw,
 				filter: self.filter,
 				svgw: self.num_obj.plot_size.width / window.devicePixelRatio,
 				strokeWidth: 0.2
 			},
 			self.opts.getBodyParams?.()
 		)
+		self.q = JSON.parse(JSON.stringify(tw.q)) //copy the q from the termwrapper to be able to change it, the tw q is readonly
```

Comment:
@siosonel is self.q part of state and thus it shouldn't be changed here (always via dispatch loop)?

Reply:
@xzhou82 `tw` is part of plot state, but the need for dispatching depends on the code step
- do not dispatch tw change when only displaying data for different mode tabs (so temporary menu UI changes)
- dispatch tw changes as state updates if the change is something that could be saved in a session (so UI menu changes typically does not belong to this)